### PR TITLE
loader, rules: use (pre)trigger.ctcp instead of tags

### PR DESCRIPTION
**Pull request by Exirel**
### Description

Sopel 8's steps of #1683:

* replace `trigger.tags['intent']` by `trigger.ctcp` (same for `PreTrigger`)
* replace `plugin_callable.intents` by `plugin_callable.ctcp`
* replace `AbstractRule.match_intent` by `AbstractRule.match_ctcp`
* update tests and documentation accordingly
* add various type hint

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

File: sopel/loader.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def is_triggerable(obj):
 
     A triggerable is a callable that will be used by the bot to handle a
     particular trigger (i.e. an IRC message): it can be a regex rule, an
-    event, an intent, a command, a nickname command, or an action command.
+    event, a CTCP command, a command, a nickname command, or an action command.
     However, it must not be a job or a URL callback.
```
Comment (dgw) [inline sopel/loader.py:143]: Would you mind rewrapping this paragraph? Can be its own commit if you want; it just annoys me now to have one line at 79c when most are 72c or less.

File: test/test_loader.py [hunk 2]
```diff
@@ -314,7 +314,7 @@ def test_clean_callable_default(tmpconfig, func):
     assert not hasattr(func, 'commands')
     assert not hasattr(func, 'nickname_commands')
     assert not hasattr(func, 'action_commands')
-    assert not hasattr(func, 'intents')
+    assert not hasattr(func, 'ctcp')
```
Comment (dgw) [inline test/test_loader.py:317]: What, you don't want to make sure nothing is adding the old `intents` attr? 😁

File: sopel/plugins/rules.py [hunk 3]
```diff
@@ -619,12 +619,12 @@ def match_event(self, event) -> bool:
         """
 
     @abc.abstractmethod
-    def match_intent(self, intent) -> bool:
-        """Tell if the rule matches this ``intent``.
+    def match_ctcp(self, command: Optional[str]) -> bool:
```
Comment (dgw) [inline sopel/plugins/rules.py:622]: Reading this before having looked at the changes to triggers, I was wondering why this would be optional. I see now that you're using `Optional` to allow the default value (`None`) of `PreTrigger.ctcp` if no CTCP command is present in the line.

One day I will get used to type annotations… I hope! 😅

File: sopel/trigger.py [hunk 4]
```diff
@@ -201,10 +206,10 @@ def __init__(
 
         # Parse CTCP into a form consistent with IRCv3 intents
         if self.event == 'PRIVMSG' or self.event == 'NOTICE':
-            intent_match = PreTrigger.intent_regex.match(self.args[-1])
-            if intent_match:
-                intent, message = intent_match.groups()
-                self.tags['intent'] = intent
+            ctcp_match = PreTrigger.ctcp_regex.match(self.args[-1])
+            if ctcp_match:
```
Comment (dgw) [inline sopel/trigger.py:210]: Is this a good time to make this check a bit more explicit?

```suggestion
            if ctcp_match is not None:
```

File: sopel/trigger.py [hunk 5]
```diff
@@ -328,8 +333,8 @@ class Trigger(str):
     .. important::
 
         Use this attribute instead of the ``intent`` tag in :attr:`tags`.
-        Message intents never made it past the IRCv3 draft stage, and Sopel will
-        drop support for them in a future release.
+        Message intents never made it past the IRCv3 draft stage, and Sopel
+        dropped support for them in Sopel 8.
     """
```
Comment (dgw) [inline sopel/trigger.py:338]: I'm gonna nitpick again, just like I did on your other PR today.

```suggestion

    """
```

**Review by dgw — COMMENTED** [review]
I've taken a look, marked a few minor things, and updated the task list in #1683. Will also mark this PR as a draft right after submitting the review, since you said on IRC that you meant to open it in the draft state.

File: test/test_loader.py (hunk 2 above)
Comment (Exirel) [reply test/test_loader.py:317]: Indeed, I don't care. 😁

**Comment by Exirel** [comment]
This PR was seriously outdated, and I wanted to be sure that was not going to add documentation that would conflict with anything new.

As a result, the doc looks pretty nice with the latest Furo update.

**Review by dgw — APPROVED** [review]
Anything I missed beyond this point will just have to become a "#2253 broke it" comment in some later issue/patch.
